src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/SourcelinkTests.cs

```diff
@@ -0,0 +1,136 @@
+    /// </summary>
+    /// <returns>Path to sourcelink tool binary.</returns>
+    private string GetSourcelinkToolPath()
+    {
+        const string SourcelinkToolPackageNamePattern = "dotnet-sourcelink*nupkg";
+        const string SourcelinkToolBinaryFilenamePattern = "dotnet-sourcelink.dll";
+
+        string toolPackageDir = Directory.CreateDirectory(Path.Combine(SourcelinkRoot, "sourcelink-tool")).FullName;
+        Utilities.ExtractTarball(Config.SourceBuiltArtifactsPath, toolPackageDir, SourcelinkToolPackageNamePattern);
+        string[] files = Directory.GetFiles(toolPackageDir, SourcelinkToolPackageNamePattern, SearchOption.AllDirectories);
+        Assert.False(files.Length > 1, "Unexpected - PSB Artifacts archive should contain only one sourcelink tool package.");
+        Assert.False(files.Length == 0, "Did not find sourcelink tool package in PSB Artifacts archive.");
+
+        string extractedToolPath = Directory.CreateDirectory(Path.Combine(toolPackageDir, "extracted")).FullName;
+        Utilities.ExtractNupkg(files[0], extractedToolPath);
+
+        files = Directory.GetFiles(extractedToolPath, SourcelinkToolBinaryFilenamePattern, SearchOption.AllDirectories);
+        Assert.False(files.Length > 1, $"Unexpected - found more than one sourcelink tool binary with filename pattern: {SourcelinkToolBinaryFilenamePattern}");
+        Assert.False(files.Length == 0, $"Did not find sourcelink tool binary with expected filename pattern: {SourcelinkToolBinaryFilenamePattern}");
+
+        return files[0];
+    }
+
+    private IEnumerable<string> GetAllSymbolsPackages()
+    {
+        /*
+            At the moment we validate sourcelinks from runtime symbols package.
+            The plan is to make symbols, from all repos, available in source-build artifacts.
+            Once that's available, this code will be modified to validate all available symbols.
+            Tracking issue: https://github.com/dotnet/source-build/issues/3612
+        */
+
+        // Runtime symbols package lives in the same directory as PSB artifacts.
+        // i.e. <repo-root>/artifacts/x64/Release/runtime/dotnet-runtime-symbols-fedora.36-x64-8.0.0-preview.7.23355.7.tar.gz
+        string runtimeSymbolsPackageNamePattern = "dotnet-runtime-symbols-*.tar.gz";
+        string[] files = Directory.GetFiles(Path.GetDirectoryName(Config.SourceBuiltArtifactsPath), runtimeSymbolsPackageNamePattern, SearchOption.AllDirectories);
+        Assert.False(files.Length > 1, "Unexpected - found more than one runtime symbols archives.");
+        Assert.False(files.Length == 0, "Did not find runtime symbols archive.");
+
+        yield return files[0];
+    }
+
+    /// <summary>
+    /// Extracts packages to subdirectories of the common symbols root directory.
+    /// </summary>
+    /// <returns>Path to common symbols root directory.</returns>
+    private string ExtractPackages(IEnumerable<string> packages)
```

Comment:
Renaming this in terms of symbols would help the readability and be more consistent with GetAllSymbolsPackages IMO.
```suggestion
    private string ExtractSymbolPackages(IEnumerable<string> packages)
```

Reply:
Thanks - will update this.

Reply:
Fixed with https://github.com/dotnet/installer/pull/17309/commits/72aacfcd5be9659d9fca751b7728cdcffb13970f